proposals/0002-spatial-navigation.md

```diff
@@ -0,0 +1,143 @@
+
+To introduce configurational flexibility, a new configuration option `enableRCUNavigation` will be introduced. This option will enable or disable the spatial navigation feature.
+
+## Bundle Size Considerations
+At this stage, we haven't precisely determined the impact of the spatial navigation feature on the bundle size. If the integration of spatial navigation results in a significant growth in the bundle size, we‘ll introduce an alternative Video.js bundle with a spatial navigation feature included.
+
+## Code and UI changes
+
+This new feature will follow these steps:
+- Introduce a public option named `enableRCUNavigation`.
+- When a player is set up with the rcu navigation feature active, initiate the SpatialNavigation instance.
+- Collect every focusable component and incorporate them into the SpatialNavigation instance.
+- Focus first available navigable candidate.
+- Once an arrow is pressed we should go through Basic Spatial Navigation Heuristics (call navigate(direction))
+  - Find Candidates
+  - Select the best candidate
+  - Focus the best candidate
+
+### In terms of changes to the code:
+
+We need to make sure that events can still move upwards when spatial navigation is turned on (`enableRCUNavigation` is set to true). Currently, it’s blocking all keydown events with `event.stopPropagation()`, except for TAB key.
+Block the UP and DOWN keys to execute `stepForward()` and `stepBack()` functions, as these keys will be needed for navigation from and to the progress bar (Slider).
+
+### In terms of adding the code
+
+Spatial-navigation interface and Component class extension
+
+```
+interface Positions {
+boundingClientRect: DOMRectReadOnly;
+center: DOMRectReadOnly;
+}
+
+interface Component {
+// Is considered as focusable component.
+getIsFocusable(): boolean;
+// Is currently visible/enabled/etc...
+getIsAvailableToBeFocused(): boolean;
+// on focus handler
+handleFocus(): void;
+// on blur handler
+handleBlur(): void;
+// on submit handler
+handleEnter(): void;
+// component's element boundingClientRect and center
+getPositions(): Positions;
+// focus component's element
+focus(): void;
+}
+
+enum SpatialNavigationDirection {
+Left = 'left',
+Right = 'right',
+Up = 'up',
+Down = 'down'
+}
+
+interface SpatialNavigation {
+// current focusable components:
+getComponents(): Set<Component>;
+// start list keydown events
+start(): void;
+// stop listen key down events
+stop(): void;
+// add focusable component
+add(component: Component): void;
+// remove focusable component
+remove(component: Component): void;
+// temporary pause spatial navigation
+pause(): void;
+// resume spatial navigation
+resume(): void;
+// clear current list of focusable components
+clear(): void;
+// run spatial navigation Heuristics (re-use https://github.com/WICG/spatial-navigation/blob/main/polyfill/spatial-navigation-polyfill.js#L147)
+move(direction: SpatialNavigationDirection): void;
+}
+```
+
+### In terms of changes the UI
```

Comment:
In order to ensure backward compatibility, we should only do this if the feature is enabled.